1-Draft/RFC00XX-Native-Command-Error-Handling.md

```diff
@@ -0,0 +1,229 @@
+- For non-zero exit codes and except for the value 'Ignore', an `ErrorRecord` will be added to
+  `$Error` that wraps the exit code and the command executed that returned the exit code.
+- Initially, only the existing values of `$ErrorActionPreference` will be supported.
+- The set of values may be extended later to include `MatchErrorActionPreference`, which should
+  apply the `$ErrorActionPreference` setting to native commands.
+- A conversion will occur between `$PSNativeCommandErrorAction` and `$ErrorActionPreference`
+  values, where `MatchErrorActionPreference` is converted to the current value of
+  `$ErrorActionPreference`
+
+Valid values for `$ErrorActionPreference`
+
+| Value           | Definition
+----------------  | -------------------
+| Break           | Enter the debugger when an error occurs or when an exception is raised.
+| Continue        | (Default) - Displays the error message and continues executing.
+| Ignore          | Suppresses the error message and continues to execute the command. The Ignore value is intended for per-command use, not for use as saved preference. Ignore isn't a valid value for the $ErrorActionPreference variable.
+| Inquire         | Displays the error message and asks you whether you want to continue.
+| SilentlyContinue| No effect. The error message isn't displayed and execution continues without interruption.
+| Stop            | Displays the error message and stops executing. In addition to the error generated, the Stop value generates an ActionPreferenceStopException object to the error stream. stream
+| Suspend         | Automatically suspends a workflow job to allow for further investigation.
+
+The reported error record should be created with the following details:
+
+| Property        | Definition
+----------------  | -------------------
+| exception:      | `ExitCode`, with the exit code of the failed command.
+| error id:       | `"Program {0} failed with unhandled exit code {1}"`, with the command name and the exit code, from resource string `ProgramFailedToComplete`.
+| error category: | `ErrorCategory.NotSpecified`.
+| object:         | exit code
+
+ This does not provide the actual semantics of bash `set -eo pipefail` as Bourne shell-style
+ integration with existing status handling syntax is not implemented.
+
+`Set -eo pipefail` is a combination of `Set -u` and `Set -o pipefail`:
+
+- `Set -u`, a reference to any variable not previously defined results in an error. Similar to
+  Set-StrictMode in PowerShell.
+- `Set -o pipefail`, by default, pipeline success is determined by the last command executed. `Set
+  -o pipefail` returns an error if any command in the pipeline fails.
+
+ As a result of not including the semantics of `Set -eo pipefail`, PowerShell script logic for
+ handling native command exit codes would need to use either `try`..`catch` or existing exit status
+ handling language constructs, according to the setting of this preference variable.
```

Comment:
I think @SteveL-MSFT really wanted this included, as it was mentioned in some of the discussion of the original RFC and issue, but I think it can probably be stashed below in the Considerations.

Reply:
Removed and placed in the considerations section.